app/controllers/projects_controller.rb

```diff
@@ -1,66 +1,72 @@
 class ProjectsController < ApplicationController
   before_action :authenticate_request, except: :index
   before_action :find_lesson
-  before_action :find_project, only: %i(update destroy)
+  before_action :find_project, only: %i(update destroy report)
 
   authorize_resource only: %i(update destroy)
 
   def index
-    @projects = projects
+    @projects = all_projects.page(params[:page]).per(50)
     @course = CourseDecorator.new(@lesson.course)
+    @user = current_user
+    @project = @projects.where(user_id: current_user.id).first if current_user
   end
 
   def create
     @project = new_project(project_params)
-    @project.save
-    set_recent_submissions
+    @project_saved = @project.save
+    @projects = latest_projects
   end
 
   def update
-    @project.update(project_params)
+    @project_updated = @project.update(project_params)
```

Comment:
Same for this one, It's not the controllers responsibility to do things like this.

I would add a new method to the ProjectDecorator

```
def updated?
   project.updated_at != project.created_at
end
```

Reply:
I am not sure if the logic to find a record is updated is right. Suppose the first update worked, in this case, `updated_at` field would get updated and won't be equal to the `created_at` field. so, the `updated?` method will return `true` as expected.

Lets say, the second attempt to update the project failed. In this case, `updated?` should return `false but ``project.updated_at != project.created_at` will return true which is a false positive.

Reply:
That would suggest a flaw in the way we are doing things. Perhaps we should rethink things in the controller action for this.